config/routes.rb

```diff
@@ -12,6 +12,7 @@
   get '/container-hosts/:id', to: 'container_hosts#show', as: 'container_host'
 
   post '/containers', to: 'containers#create', as: 'containers_create'
+  post '/recreate_container', to: 'containers#recreate', as: 'containers_recreate'
```

Comment:
Why no have Restful routes?  It could be something like /containers/:id/recreate.

Reply:
we're following the pattern in the other API: `DELETE '/containers'` that was already created.

it doesn't use `id`, instead it accept hostname (alphanumeric).

Reply:
Yes, because we are using hostnames as our IDs. This data stays with LXD db. And with Clustering it is sure that hostnames would be unique.

Reply:
so here we can say /containers/recreate and use hostname similar to the delete api. Or I think /containers/:hostname/recreate. Thoughts @giosakti @kumaranvram ?

Reply:
in my opinion we should keep it consistent with other API and avoid RESTful for now..

the problem with alphanumeric characters as part of restful url such as `/containers/:hostname/recreate` is that if user do something weird like renaming their hostname to `recreate` then it would be problematic (we have to implement stringent hostname blacklist)

Reply:
that was a bad example, let me give you another case of problematic situation.

consider we have, new method called `action_a` and we have routes like these:
```
POST /containers/action_a
POST /containers/:hostname/recreate
```

and the user somehow named their container `action_a`.. then nobody can access `POST /containers/:hostname/recreate` for that particular container

Reply:
But it can `POST /containers/recreate` with container name in body. The resource remains containers. But agree we will have to do something about profile crud, maybe move :name to body there as well.

Reply:
keeping it similar to existing apis would also mean /containers/recreate. similar to delete.

Reply:
ah yes, you're right, will refactor this and get back to you